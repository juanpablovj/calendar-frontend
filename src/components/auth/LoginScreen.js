import React from "react";
import { useDispatch } from "react-redux";
import Swal from "sweetalert2";
import { startLogin, startRegister } from "../../actions/auth";
import { useForms } from "../../hooks/useForms";
import "./login.css";

export const LoginScreen = () => {
  const dispatch = useDispatch();

  const [formLoginValues, handleLoginInputChange] = useForms({
    loginEmail: "jp@jp.com",
    loginPassword: "123123",
  });

  const [formRegisterValues, handleRegisterInputChange] = useForms({
    registerName: "Juanito",
    registerEmail: "jp@jp2.com",
    registerPassword1: "123123",
    registerPassword2: "123123",
  });

  const { registerName, registerEmail, registerPassword1, registerPassword2 } =
    formRegisterValues;

  const { loginEmail, loginPassword } = formLoginValues;

  const handleSubmitLogin = (e) => {
    e.preventDefault();
    dispatch(startLogin(loginEmail, loginPassword));
  };

  const handleSubmitRegister = (e) => {
    e.preventDefault();

    if (registerPassword1 !== registerPassword2) {
      return Swal.fire(
        "Error",
        "Las contraseñas deben de ser iguales",
        "error"
      );
    }

    dispatch(startRegister(registerName, registerEmail, registerPassword1));
  };

  return (
    <div className="container login-container">
      <div className="row">
        <div className="col-md-6 login-form-1">
          <h3>Ingreso</h3>
          <form onSubmit={handleSubmitLogin}>
            <div className="form-group">
              <input
                type="email"
                className="form-control mb-3"
                placeholder="Correo"
                name="loginEmail"
                value={loginEmail}
                onChange={handleLoginInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="text"
                className="form-control mb-3"
                placeholder="Password"
                name="loginPassword"
                value={loginPassword}
                onChange={handleLoginInputChange}
              />
            </div>
            <div className="form-group">
              <input type="submit" className="btnSubmit mb-3" value="Login" />
            </div>
          </form>
        </div>

        <div className="col-md-6 login-form-2">
          <h3>Registro</h3>
          <form onSubmit={handleSubmitRegister}>
            <div className="form-group">
              <input
                type="text"
                className="form-control mb-3"
                placeholder="Nombre"
                name="registerName"
                value={registerName}
                onChange={handleRegisterInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="email"
                className="form-control mb-3"
                placeholder="Correo"
                name="registerEmail"
                value={registerEmail}
                onChange={handleRegisterInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control mb-3"
                placeholder="Contraseña"
                name="registerPassword1"
                value={registerPassword1}
                onChange={handleRegisterInputChange}
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                className="form-control mb-3"
                placeholder="Repita la contraseña"
                name="registerPassword2"
                value={registerPassword2}
                onChange={handleRegisterInputChange}
              />
            </div>

            <div className="form-group">
              <input
                type="submit"
                className="btnSubmit mb-3"
                value="Crear cuenta"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};
