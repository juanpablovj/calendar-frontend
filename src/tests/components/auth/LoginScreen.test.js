import React from "react";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom";
import { LoginScreen } from "../../../components/auth/LoginScreen";
import { startLogin, startRegister } from "../../../actions/auth";
import Swal from "sweetalert2";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <LoginScreen />
  </Provider>
);

jest.mock("../../../actions/auth", () => ({
  startLogin: jest.fn(),
  startRegister: jest.fn(),
}));

jest.mock("sweetalert2", () => ({
  fire: jest.fn(),
}));

describe("Pruebas de LoginScreen", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("should mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should de llamar el dispatch del login", () => {
    wrapper.find('input[name="loginEmail"]').simulate("change", {
      target: {
        name: "loginEmail",
        value: "jpcito",
      },
    });
    wrapper.find('input[name="loginPassword"]').simulate("change", {
      target: {
        name: "loginPassword",
        value: "123123",
      },
    });

    wrapper.find("form").at(0).prop("onSubmit")({ preventDefault() {} });
    expect(startLogin).toHaveBeenCalledWith("jpcito", "123123");
  });

  test("no hay registro si las contraseñas son diferentes", () => {
    wrapper.find('input[name="registerPassword1"]').simulate("change", {
      target: {
        name: "registerPassword1",
        value: "123123",
      },
    });
    wrapper.find('input[name="registerPassword2"]').simulate("change", {
      target: {
        name: "registerPassword2",
        value: "123124",
      },
    });

    wrapper.find("form").at(1).prop("onSubmit")({ preventDefault() {} });
    expect(startRegister).not.toHaveBeenCalled();
    expect(Swal.fire).toHaveBeenCalledWith(
      "Error",
      "Las contraseñas deben de ser iguales",
      "error"
    );
  });

  test("Registro con contraseñas iguales", () => {
    wrapper.find('input[name="registerPassword1"]').simulate("change", {
      target: {
        name: "registerPassword1",
        value: "12345",
      },
    });
    wrapper.find('input[name="registerPassword2"]').simulate("change", {
      target: {
        name: "registerPassword2",
        value: "12345",
      },
    });

    wrapper.find("form").at(1).prop("onSubmit")({ preventDefault() {} });
    expect(startRegister).toHaveBeenCalledWith(
      "Juanito",
      "jp@jp2.com",
      "12345"
    );
    expect(Swal.fire).not.toHaveBeenCalled();
  });
});
