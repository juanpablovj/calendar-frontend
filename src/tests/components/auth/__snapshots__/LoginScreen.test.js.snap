// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de LoginScreen should mostrarse correctamente 1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1"
    >
      <h3>
        Ingreso
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control mb-3"
            name="loginEmail"
            onChange={[Function]}
            placeholder="Correo"
            type="email"
            value="jp@jp.com"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control mb-3"
            name="loginPassword"
            onChange={[Function]}
            placeholder="Password"
            type="text"
            value="123123"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit mb-3"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
    <div
      className="col-md-6 login-form-2"
    >
      <h3>
        Registro
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control mb-3"
            name="registerName"
            onChange={[Function]}
            placeholder="Nombre"
            type="text"
            value="Juanito"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control mb-3"
            name="registerEmail"
            onChange={[Function]}
            placeholder="Correo"
            type="email"
            value="jp@jp2.com"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control mb-3"
            name="registerPassword1"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
            value="123123"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control mb-3"
            name="registerPassword2"
            onChange={[Function]}
            placeholder="Repita la contraseña"
            type="password"
            value="123123"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit mb-3"
            type="submit"
            value="Crear cuenta"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
